### PR TITLE
feat: refactor onboarding

### PR DESCRIPTION
**Pull request by mohandast52**
## Proposed changes

- Onboarding flow 

(The app is running after commenting out a lot of code.)

Screenshot

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: frontend/components/SetupPage/Create/SetupCreateSafe.tsx [hunk 1]
```diff
@@ -22,47 +22,92 @@ const capitalizedMiddlewareChainNames = {
...
-  const { updateMasterSafeOwners } = useMultisig();
+  const { masterSafes, refetch: updateWallets } = useMasterWalletContext();
+  // const { updateMasterSafeOwners } = useMultisig();
   const { backupSigner } = useSetup();
 
+  const masterSafeAddress = useMemo(() => {
+    if (!masterSafes) return;
+    const master = masterSafes.find(
+      (safe) => safe.owner === WalletOwnerType.Master,
+    );
```
Comment (truemiller) [inline frontend/components/SetupPage/Create/SetupCreateSafe.tsx:67]: masterSafes is pre-filtered
```suggestion
    const master = masterSafes[0];
```

File: frontend/components/SetupPage/Create/SetupEoaFunding.tsx [hunk 2]
```diff
@@ -80,7 +88,7 @@ const SetupEoaFundingWaiting = ({ chainName }: SetupEoaFundingWaitingProps) => {
         </Flex>
 
         <span className="can-select-text break-word">
-          {`ETH: ${masterEoaAddress}`}
+          {`ETH: ${masterEoaAddress || NA}`}
```
Comment (truemiller) [inline frontend/components/SetupPage/Create/SetupEoaFunding.tsx:91]: will pull `homeChainId` from `selectedAgentConfig` in `useServices` in future, then `getNativeTokenSymbol(homeChainId)` to get native gas token

not needed for now, but can hardcode XDAI instead for now

```suggestion
          {`XDAI: ${masterEoaAddress || NA}`}
```

File: frontend/components/SetupPage/Create/SetupEoaFunding.tsx [hunk 3]
```diff
@@ -141,37 +141,46 @@ export const SetupEoaFundingForChain = ({
...
+    chainConfig: CHAIN_CONFIG[ChainId.Gnosis],
+    requiredEth: MIN_ETH_BALANCE_THRESHOLDS[ChainId.Gnosis].safeCreation,
   },
 };
 
 export const SetupEoaFunding = () => {
   const { goto } = useSetup();
-  const { masterEoaAddress } = useWallet();
+  const { masterEoa } = useMasterWalletContext();
+  const { walletBalances } = useBalanceContext();
```
Comment (truemiller) [inline frontend/components/SetupPage/Create/SetupEoaFunding.tsx:181]: can get `masterEoaBalances` from `useMasterBalances()`

File: frontend/components/SetupPage/Create/SetupEoaFunding.tsx [hunk 4]
```diff
@@ -225,6 +234,12 @@ export const SetupEoaFunding = () => {
     goto(SetupScreen.SetupCreateSafe);
   }, 5000);
 
+  const eoaBalance = walletBalances.find((e) => e.isNative);
```
Comment (truemiller) [inline frontend/components/SetupPage/Create/SetupEoaFunding.tsx:237]:
```suggestion
  const eoaBalance = walletBalances.find((e) => e.isNative); // TODO: ensure we pull correct native with chainId when implementing multi-chain
```

File: frontend/components/SetupPage/Create/SetupEoaFunding.tsx [hunk 5]
```diff
@@ -225,6 +234,12 @@ export const SetupEoaFunding = () => {
...
+  const isFunded =
+    eoaBalance?.chainId === ChainId.Ethereum &&
```
Comment (truemiller) [inline frontend/components/SetupPage/Create/SetupEoaFunding.tsx:239]:
```suggestion
    eoaBalance?.chainId === ChainId.Gnosis &&
```

File: frontend/components/SetupPage/Create/SetupEoaFunding.tsx (hunk 4 above)
Comment (Tanya-atatakai) [inline frontend/components/SetupPage/Create/SetupEoaFunding.tsx]: should we check here chainId as well? like the `currentChain` or maybe even `homeChainId` from selectedAgentConfig?

+ probably can use useMasterBalances instead

File: frontend/components/SetupPage/Create/SetupEoaFunding.tsx (hunk 5 above)
Comment (Tanya-atatakai) [inline frontend/components/SetupPage/Create/SetupEoaFunding.tsx]: why check ChainId.Ethereum? should be gnosis?

File: frontend/config/mechs.ts [hunk 6]
```diff
@@ -24,14 +25,18 @@ export const MECHS: Mechs = {
       name: 'Agent Mech',
       contract: new MulticallContract(
         '0x77af31De935740567Cf4fF1986D04B2c964A786a',
-        AGENT_MECH_ABI,
+        AGENT_MECH_ABI.filter(
```
Comment (truemiller) [inline frontend/config/mechs.ts:28]: probably better to put on the ABI itself, but this is fine for now; no need for JsonFragment either, can use Abi from types

File: frontend/components/SetupPage/SetupWelcome.tsx [hunk 7]
```diff
@@ -129,9 +130,15 @@ export const SetupWelcomeLogin = () => {
...
-  const { isBalanceLoaded, masterEoaBalance: eoaBalance } = useBalanceContext();
+  const { selectedService } = useServices();
+  const { masterSafes, masterWallets: wallets } = useMasterWalletContext();
+  const { isLoaded, walletBalances } = useBalanceContext();
+
+  const masterSafe =
+    masterSafes?.find(
+      (safe) => safe.chainId === selectedService?.home_chain_id,
+    ) ?? null;
+  const eoaBalanceEth = walletBalances?.find((balance) => balance.isNative);
```
Comment (Tanya-atatakai) [inline frontend/components/SetupPage/SetupWelcome.tsx:141]: unsure it shouldn't check chainId as well, and also maybe better use `useMasterBalances`, worth checking with @truemiller

File: frontend/pages/_app.tsx [hunk 8]
```diff
@@ -38,37 +38,39 @@ export default function App({ Component, pageProps }: AppProps) {
...
+                  <StakingProgramProvider>
+                    <StakingContractDetailsProvider>
+                      <RewardProvider>
+                        <BalanceProvider>
+                          <SetupProvider>
+                            <SettingsProvider>
+                              <ConfigProvider theme={mainTheme}>
+                                <ModalProvider>
+                                  {isMounted ? (
+                                    <QueryClientProvider client={queryClient}>
```
Comment (Tanya-atatakai) [inline frontend/pages/_app.tsx:54]: now we have two QueryClientProvider it seems

File: frontend/pages/_app.tsx [hunk 9]
```diff
@@ -38,37 +38,39 @@ export default function App({ Component, pageProps }: AppProps) {
...
+                                      <SystemNotificationTriggers>
+                                        <Layout>
+                                          <Component {...pageProps} />
+                                        </Layout>
+                                      </SystemNotificationTriggers>
+                                    </QueryClientProvider>
```
Comment (truemiller) [inline frontend/pages/_app.tsx:60]: good catch here, hadn't even thought about moving query provider!
```suggestion
                                    
                                      <SystemNotificationTriggers>
                                        <Layout>
                                          <Component {...pageProps} />
                                        </Layout>
                                      </SystemNotificationTriggers>
                                    
```

File: frontend/components/SetupPage/SetupWelcome.tsx (hunk 7 above)
Comment (truemiller) [reply frontend/components/SetupPage/SetupWelcome.tsx:141]: yes, we should, have suggested TODO comment here: https://github.com/valory-xyz/olas-operate-app/pull/478#discussion_r1852704485

not a major issue for getting predict only version for now though

**Review by Tanya-atatakai — APPROVED** [review]

**Review by truemiller — APPROVED** [review]

**Pull request merged**